src/main/java/raphael/parser/Parser.java

```diff
@@ -0,0 +1,103 @@
+package raphael.parser;
+
+import raphael.command.Command;
+import raphael.command.DeleteCommand;
+import raphael.command.ExitCommand;
+import raphael.command.FindCommand;
+import raphael.command.CheckCommand;
+import raphael.command.EditCommand;
+import raphael.command.AddCommand;
+import raphael.command.ListCommand;
+import raphael.exception.RaphaelException;
+import raphael.task.Todo;
+import raphael.task.Event;
+import raphael.task.Deadline;
+import raphael.task.Task;
+
+/**
+ * Parses the command line input
+ */
+public class Parser {
+    /**
+     * Returns a command object after interpreting the input received from the command line. This method will throw
+     * exception if the input does not fulfill the required format.
+     *
+     * @param input the input string read from command line
+     * @return a command that corresponding to the keyword entered
+     * @throws RaphaelException exception exclusive to Raphael
+     */
+    public static Command parse(String input) throws RaphaelException {
+        if (input.isEmpty()) {
+            throw new RaphaelException("The command can't be empty!");
+        }
+        final String[] inputArr = input.split(" ", 2);
+        switch(inputArr[0]) {
+            case "bye":
```

Comment:
Not sure if the case in switch case should be further indented or on the same line according to 2103's coding standard?